packages/blaze/template.js

```diff
@@ -309,6 +311,43 @@ Blaze.TemplateInstance.prototype.autorun = function (f) {
 };
 
 /**
+ * @summary A version of [Meteor.subscribe](#meteor_subscribe) that is stopped
+ * when the template is destroyed.
+ * @return {SubscriptionHandle} The subscription handle to the newly made
+ * subscription. Call `handle.stop()` to manually stop the subscription, or
+ * `handle.ready()` to find out if this particular subscription has loaded all
+ * of its inital data.
+ * @locus Client
+ * @param {String} name Name of the subscription.  Matches the name of the
+ * server's `publish()` call.
+ * @param {Any} [arg1,arg2...] Optional arguments passed to publisher function
+ * on server.
+ * @param {Function|Object} [callbacks] Optional. May include `onError` and
+ * `onReady` callbacks. If a function is passed instead of an object, it is
+ * interpreted as an `onReady` callback.
+ */
+Blaze.TemplateInstance.prototype.subscribe = function (/* arguments */) {
+  var subHandle = this.view.subscribe.apply(this.view, arguments);
+
+  this._subscriptionHandles.set(
+    this._subscriptionHandles.get().concat([subHandle]));
+
+  return subHandle;
+};
+
+/**
+ * @summary A reactive function that returns true when all of the subscriptions
+ * called with [this.subscribe](#TemplateInstance-subscribe) are @ready.
+ * @return {Boolean} True if all subscriptions on this template instance are
+ * ready.
+ */
+Blaze.TemplateInstance.prototype.ready = function () {
```

Comment:
This could break code that uses a `ready` property on the template instance.  It doesn't break the Discover Meteor code, which sets `this.ready = new ReactiveVar` on the template instance (because the inherited property will be shadowed), but it would break someone's code who checks `this.ready` but never explicitly initializes it to `false` (since now `this.ready` starts out truthy).

I think I'm ok with that, but we should at least have a unit test that verifies that shadowing `this.ready()` works.  (I don't see any tests of `this.ready()` actually, only `Template.ready`.)  That way, we won't accidentally break apps that rely on being able to set `this.ready` (like apps following Discover Meteor's pattern) at some point in the future if we change how the template instance object works.

If someone shadows `this.ready`, I also note that it will break `Template.ready`.  But we could decide we don't care about that.